clients/include/testing_axpy.hpp

```diff
@@ -17,6 +17,15 @@
 
 using namespace std;
 
+#define CLEANUP()                                                  \
```

Comment:
You can specialize std::unique_ptr<> with a custom deleter, to call hipFree.  You can google that for syntax.